OZON.ru.js

```diff
@@ -9,596 +9,130 @@
-			return "4";
-		case "пятое":
-			return "5";
-		case "шестое":
-			return "6";
-		case "седьмое":
-			return "7";
-		case "восьмое":
-			return "8";
-		case "девятое":
-			return "9";
-		case "десятое":
-			return "10";
-		default:
-			return numStr;
-		}
-	}
-	return "";
-}
+function getAdditionalProperty(element) {
+	var matches = /[А-я]+(?=:)/.exec(element.textContent);
+	if (matches === null)
+		return undefined;
+	var propertyName = matches[0];
+	var propertyValue = xpath('string(./a/@title)', element).stringValue;
 
-function ParseProductDetail(pdXPathRes) {
-	var result = new Object();
-	var creatorsObject = new Object();
-	var pdDOMNode;
-	while (pdDOMNode = pdXPathRes.iterateNext()) {
-		var text = CleanText(pdDOMNode.textContent);
-		var name = text.replace(/[:,\s].*$/, "");
-		var value = text.replace(/^\S*\s+/, "");
-		//Zotero.debug(name + " -> " + value);
-		switch (name) {
-		case "Автор":
-		case "Авторы":
-			creatorsObject["authors"] = GetCreators(value, "author");
-			break;
-		case "Составитель":
-			creatorsObject["compilers"] = GetCreators(value, "editor");
-			break;
-		case "Переводчик":
-		case "Переводчики":
-			creatorsObject["translators"] = GetCreators(value, "translator");
-			break;
-		case "Редактор":
-		case "Редакторы":
-			creatorsObject["editors"] = GetCreators(value, "editor");
-			break;
-		case "Язык":
-		case "Языки":
-			result["language"] = GetLanguage(value);
-			break;
-		case "Издательство":
-			result["publisher"] = value;
-			break;
-		case "Серия":
-			result["series"] = value;
-			break;
-		case "ISBN":
-			result["ISBN"] = GetISBN(value);
-			result["date"] = GetDate(value);
-			break;
-		default:
-			//"Naked" year
-			if (name.match(/^\d\d\d\d$/) && !result["date"]) {
-				result["date"] = name;
-			}
-		}
-	}
-	creatorsObject["all"] = new Array();
-	if(creatorsObject["authors"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["authors"]);
-	}
-	if(creatorsObject["compilers"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["compilers"]);
-	}
-	if(creatorsObject["translators"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["translators"]);
+	return {
+		name: propertyName,
+		value: propertyValue
 	}
-	if(creatorsObject["editors"]) {
-		creatorsObject["all"] =
-		creatorsObject["all"].concat(creatorsObject["editors"]);
-	}
-	if(creatorsObject["all"]) {
-		result["creators"] = creatorsObject["all"];
-	}
-	return result;
 }
 
-function ParseTechDesc(nameXPathRes, valueXPathRes) {
-	var result = Object();
-	var nameDOMNode;
-	var valueDOMNode;
-	while( (nameDOMNode = nameXPathRes.iterateNext()) &&
-	(valueDOMNode = valueXPathRes.iterateNext()) ) {
-		var nameText = CleanText(nameDOMNode.textContent);
-		var valueText = CleanText(valueDOMNode.textContent);
-		if (nameText == "Страниц") {
-			result["numPages"] = GetNumPages(valueText);
-			return result;
-		}
+function splitFullName(fullName) {
```

Comment:
It doesn't work for input like "Henry S. Warren jr." (my function doesn't either). I can submit an issue if needed.